src/sampler.jl

```diff
@@ -1,3 +1,28 @@
+# Update of hamiltonian and proposal
+
+Parameters.reconstruct(h::Hamiltonian, ::AbstractAdaptor) = h
+function Parameters.reconstruct(h::Hamiltonian, adaptor::Union{MassMatrixAdaptor, CompositeAdaptor})
+    metric = renew(h.metric, getM⁻¹(adaptor))
+    return reconstruct(h, metric=metric)
+end
+
+Parameters.reconstruct(τ::AbstractProposal, ::AbstractAdaptor) = τ
+function Parameters.reconstruct(τ::AbstractProposal, adaptor::Union{StepSizeAdaptor, CompositeAdaptor})
+    # FIXME: this does not support change type of `ϵ` (e.g. Float to Vector)
+    # FIXME: this is buggy for `JitteredLeapfrog`
+    integrator = reconstruct(τ.integrator, ϵ=getϵ(adaptor))
```

Comment:
Maybe add `Parameters` prefix here for consistency.

Reply:
I think a better way to resolve this consistency of usage is by `import reconstruct` explictly in the beginning of AdvancedHMC. So that we know all the functions we extend. How do you think?

Reply:
Either way works, although `Parameters.reconstruct` is a bit more explicit about `reconstruct` is not an AHMC function.

Reply:
I agree and that's why it is written as it ATM. The convention is
- For functions to extend, we use `Module.fn` explictly
- For functions to use (those indicated by `using Module: fn1, fn2 ...` in the beginning of AHMC), we drop `Module`

However, as you point here, it leads to this inconsistency. I would prefer explicity `import` to solve it, just so it is consistency with explict `using`.

Reply:
Sounds good.